modules/optimization/doc/content/modules/optimization/getting_started/InvOptTheory.md

```diff
@@ -0,0 +1,376 @@
+
+[comment2]: <> (% ----------------------------------------------------------------------------------------------------------%)
+
+## PDE Derivatives for Inversion id=sec:PDEDerivs
+
+In this section we will present derivatives for steady state heat conduction [eqn:optimization_problem] with respect to the force or material parameters.  For all of these examples, measurement data is taken at specific locations where the objective function can be represented by [eqn:objective_integral].  We will present the discrete forms of our PDE and its derivative which most closely matches the implementation that will be used in MOOSE.
+
+The discrete form of the PDE constraint for steady state heat conduction in [eqn:optimization_problem], $\hat{g}$, is given as
+
+\begin{equation}\label{eq:discretePDE}
+\hat{\text g}=\textbf{K}\hat{\text T} - \hat{\text f}=0,
+\end{equation}
+where $\textbf{K}$ is the Jacobian matrix, $\hat{\text T}$ and $\hat{\text f}$ are the discretized  temperature and residual vectors. Element-wise definitions of the terms are
+\begin{equation} \label{eq:elementwise_discretized_terms}
+\begin{aligned}
+    \textbf{K}^{\alpha\beta}&=\sum_{e}\int\nabla^\top \text{N}^{\alpha} \cdot \kappa\cdot\nabla\text{N}^{\beta}~\text{d}\Omega,\\
+    \hat{\text f}^{\alpha} & = \sum_{e} \int N^{\alpha} f_b \text{d}\Omega + \sum_{e} \int N^{\alpha}G(T) ~\text{d}\Gamma_R,
+\end{aligned}
+\end{equation}
+where $N^{\alpha}$ denotes the finite element shape function at node $\alpha$. The solution can be expressed as $T(x)\approx \sum_{\alpha} \text{N}^{\alpha}\hat{\text T}^{\alpha} = N \hat{\text T}$. Note here the $\hat{\text f}$ includes the contribution from both the body load term ($f_b$ in [eqn:optimization_problem]) and boundary conditions (see [eqn:optimization_bcs]). We are assuming a Galerkin form for our discretized PDE by making our test and trial functions identical (both are $\text{N}^{\alpha}$). Note in the rest of this document, we omit the superscript of the shape function (i.e., $(\cdot)^{\alpha}$) and the summation over all the elements (i.e., $\sum_e$) for simplicity.
+
+To compute the derivative of PDE with respect to the design parameters, i.e., $\partial g /\partial \boldsymbol{p}$, it can be seen from [eq:elementwise_discretized_terms] that the derivatives can come from either $\textbf{K}$ or $\hat{\text f}$. For problems that have design parameters that are embedded in $f_b$ and/or the Neumann boundary condition in $G(T)$, we call them \textbf{force inversion} problems, since the derivative only depends on the body load. For design parameters that are embedded in $\textbf{K}$ and/or the convection boundary condition in $G(T)$, indicating dependence on the material property, we call them \textbf{material inversion} problems. Derivative calculation of different force inversion and material inversion problems are included in the following subsections:
```

Comment:
\textbf{force inversion} => +force inversion+ (the MooseDocs syntax).

Reply:
done